### ISSUE TITLE
Lorentz factor is wrong with inclined rotation axis

### ISSUE THREAD

**Issue by MACarlsen** [FABLE-3DXRD/xrd_simulator#29]
The function implemented for the Lorentz-factor gives a wrong results when the rotation-axis is not orthogonal to the incident beam direction (like in a laminography-experiment). Most importantly the :math:`\\eta=0` divergence has disappeared. The error is also in the paper. The correct expression is given eg. in https://journals.iucr.org/paper?a11597

```
def _lorentz(
    k_in: torch.Tensor, k_out: torch.Tensor, rot_axis: torch.Tensor
) -> torch.Tensor | float:
    """Compute the Lorentz intensity factor for all reflections.

    This function calculates the Lorentz factor for X-ray diffraction based on
    the incident beam direction, scattered wave vectors, and rotation axis.

    Parameters
    ----------
    k_in : torch.Tensor
        Incident beam direction vector with shape (3,)
    k_out : torch.Tensor
        Scattered wave vectors with shape (N, 3) or (3,)
    rot_axis : torch.Tensor
        Rotation axis vector with shape (3,)

    Returns
    -------
    torch.Tensor | float
        Lorentz factors for each reflection with shape (N,) or a single float.
        Returns infinity for geometrically impossible reflections.

    Examples
    --------
    >>> k_in = torch.tensor([1., 0., 0.])
    >>> k_out = torch.tensor([[0.5, 0.5, 0.], [0.2, 0.3, 0.1]])
    >>> rot_axis = torch.tensor([0., 0., 1.])
    >>> lorentz(k_in, k_out, rot_axis)
    tensor([1.1547, 1.0758])

    Notes
    -----
    The Lorentz factor is calculated as :math:`1/|\\mathbf{\\omega}\\cdot\\mathbf{k}_{\\mathrm{in}}\\times\\mathbf{q}|`,
    where :math:`\\mathbf{\\omega}` is the the rotation axis and :math:`\\mathbf{q}` is the scattering plane normal.
    """

    
    kp = k_out.reshape(-1, 3) if k_out.dim() == 1 else k_out

    # Normalize k for dot product calculation
    k_norm = torch.linalg.norm(k_in)
    q = kp - k_in[None :]
    l = torch.matmul(torch.cross(k_in, rot_axis), q.T)

    return k_norm**2/torch.abs(l)
```

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/aefea3d7-71a3-4542-95a1-95c63b8e0121" />
